sdk/include/platform/sunburst/platform-spi.hh

```diff
@@ -0,0 +1,403 @@
+#pragma once
+#include <cdefs.h>
+#include <debug.hh>
+#include <stdint.h>
+#include <utils.hh>
+
+namespace SonataSpi
+{
+	/// Sonata SPI Interrupts
+	typedef enum [[clang::flag_enum]]
+	: uint32_t{
+	    /// Raised when a SPI operation completes and the block has become idle.
+	    InterruptComplete = 1 << 4,
+	    /// Asserted whilst the transmit FIFO level is at or above the
+	    /// watermark.
+	    InterruptTransmitWatermark = 1 << 3,
+	    /// Asserted whilst the transmit FIFO is empty.
+	    InterruptTransmitEmpty = 1 << 2,
+	    /// Asserted whilst the receive FIFO level is at or below the watermark.
+	    InterruptReceiveWatermark = 1 << 1,
+	    /// Asserted whilst the receive FIFO is full.
+	    InterruptReceiveFull = 1 << 0,
+	  } Interrupt;
```

Comment:
Er, perhaps this is just a polarity thing, but I'd expect the `InterruptTransmitWatermark` to fire when the transmit FIFO is _below_ its watermark (that is, it has room to accept a positive natural number of bytes before overflow) and, dually, `InterruptReceiveWatermakr` to fire when the receive FIFO is _above_ its watermark (that is, it can source at least watermark-many bytes).

Reply:
Good catch! You're spot on.

- https://github.com/lowRISC/sonata-system/blob/736837cda6de2d7b6dbbe86a513d18d2eb4155e2/rtl/ip/spi/rtl/spi.sv#L168
- https://github.com/lowRISC/sonata-system/blob/736837cda6de2d7b6dbbe86a513d18d2eb4155e2/rtl/ip/spi/rtl/spi.sv#L165